content/gocourse-2019-05-melbourne.slide

```diff
@@ -1142,6 +1142,35 @@ Rob Pike's Talk ([[https://blog.golang.org/toward-go2][Towards Go 2]])
 Dave Cheney Blog ([[https://dave.cheney.net/paste/the-past-present-and-future-of-go2.pdf][Past Present and Future of Go]])
 
 
+* Go for Java devs
+
+- Functions are lambdas
+- Typed constants are enums
+- exported = public
+- unexported = protected, not private
+- struct != class
+An empty struct is pointless
+
+- embedding = composition + inheritence
+- map[type]struct{} = Set<type>
+- Beware of pointers
+- Pointers will also save your day
```

Comment:
Again, could you maybe either add an example on why and/or speaker notes please?

Reply:
Done